### ISSUE TITLE
Expose a ini file for edit online in Collabora Office / LibreOffice

### ISSUE THREAD

**Issue by juliusknorr** [nextcloud/desktop#4880]
The idea is to have a button inside of Collabora Online / LibreOffice that allows the user to switch to editing online by clicking it.

Requirement is that richdocuments would implement https://github.com/nextcloud/richdocuments/issues/2418
- Provide an endpoint with a path parameter to open the file for the user
- Expose the endpoint URL through capabilities, so that it can be picked up by the client

The desktop client would expose a `.editonline.ini` (naming could change) file in the root sync folder which contains the edit url as well as the userid and relative path, which can then be read by Collabora Office.

- [ ] Fetch the endpoint from the capabilities
- [ ] For each sync folder put a .ini file in the root, containing the format below

### Example 1

The Nextcloud root is synced to ~/Nextcloud, there would be a ~/Nextcloud/.editonline.ini:

```
[editserver]
    endpoint = https://nextcloud.example.com/index.php/apps/richdocuments/editfromlocal
    userid = admin
    relativepath = /
```

Considering a file Document.odt is opened, Collabora Office could then construct url to open in the browser:
https://nextcloud.example.com/index.php/apps/richdocuments/editfromlocal?user=admin&path=/Document.odt

### Example 2

The Nextcloud subdirectory /Engineering is synced to ~/Nextcloud, there would be a ~/Nextcloud/.editonline.ini:

```
[editserver]
    endpoint = https://nextcloud.example.com/index.php/apps/richdocuments/editfromlocal
    userid = admin
    relativepath = /Engineering
```

Considering a file Document.odt is opened, Collabora Office could then construct url to open in the browser:
https://nextcloud.example.com/index.php/apps/richdocuments/editfromlocal?user=admin&path=/Engineering/Document.odt

**Comment by juliushaertl** [comment]
@allexzander @mmeeks @tml1024 please have a quick sanity check on the summary :)

**Comment by juliushaertl** [comment]
cc @tobiasKaminsky

**Comment by mmeeks** [comment]
Looks great to me - from a naming perspective I think .editonline could perhaps be more generalized - I expect there are other things that this could be used for if it is there =) but - up to you ! Thanks.

**Comment by allexzander** [comment]
Yes, looks great @juliushaertl . I assume the `userid` is the first part of e.g. `admin@example.cloud.com`? As we also have another type of id on the server that looks like a hash (Web Dav ID, or smth.). 

@mmeeks 
As for the generic name, I propose to have it named `.external_apps_metadata.ini` or just `.metadata.ini`.

**Comment by juliushaertl** [comment]
Yes, the user id would be admin in your example.

**Comment by juliushaertl** [comment]
richdocuments implementation is there in https://github.com/nextcloud/richdocuments/pull/2419 just might need a bit more polishing, but general functionality works.

**Comment by mmeeks** [comment]
I'd go for .metadata.ini and expect that when it is there it will turn out to be useful for lots of other loosely coupled things =)

**Comment by mmeeks** [comment]
We will eventually need this; as well as a socket to talk to a remote richdocuments from a live app here I think.

**Comment by juliusknorr** [comment]
Yes, i think this would be a great enhancement still to integrate more nicely into the Collabora Office desktop app or libreoffice.

**Comment by Rello** [comment]
> Yes, i think this would be a great enhancement still to integrate more nicely into the Collabora Office desktop app or libreoffice.

pinging again: still valid? ;-)

**Comment by mmeeks** [comment]
Yes, valid :-)
